apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,319 @@ package managedidentity
 
 import (
 	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/errors"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/mock"
+)
+
+const (
+	// test Resources
+	resource              = "https://demo.azure.com"
+	resourceDefaultSuffix = "https://demo.azure.com/.default"
+
+	token = "fakeToken"
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+type SuccessfulResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpiresOn   int64  `json:"expires_on"`
+	Resource    string `json:"resource"`
+	TokenType   string `json:"token_type"`
+	ClientID    string `json:"client_id"`
+}
+
+type ErrorRespone struct {
+	Err  string `json:"error"`
+	Desc string `json:"error_description"`
+}
 
-	if err != nil {
-		return Client{}, err
+type response struct {
+	body     []byte
+	callback func(*http.Request)
+	code     int
+}
+
+func getSuccessfulResponse(resource string) ([]byte, error) {
+	expiresOn := time.Now().Add(1 * time.Hour).Unix()
+	response := SuccessfulResponse{
```

Comment:
You want the mock tests to be realistic, so you may as well start now by parameterizing this for the source and returning a response having exactly the same body fields as a real one